core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -358,7 +358,7 @@ class LogSegment private[log] (val log: FileRecords,
 
         if (batch.magic >= RecordBatch.MAGIC_VALUE_V2) {
           leaderEpochCache.foreach { cache =>
-            if (batch.partitionLeaderEpoch > cache.latestEpoch()) // this is to avoid unnecessary warning in cache.assign()
+            if (batch.partitionLeaderEpoch > cache.latestEpoch.getOrElse(-1)) // this is to avoid unnecessary warning in cache.assign()
```

Comment:
Another way to write this:
```scala
cache.latestEpoch.forall(batch.partitionLeaderEpoch > _)
```

Also, can we drop the parethesis from the definition of `latestEpoch` in `LeaderEpochCache`?

Reply:
Does that actually result in clearer code? I find that harder to read than the above

Dropped the parenthesis 👍

Reply:
It's a bit subjective for sure, so feel free to ignore the suggestion. I just thought the use of -1 was a little odd. A more explicit alternative might be something like this:
```
if (latestEpoch.isEmpty || batch.partitionLeaderEpoch > latestEpoch.get)
```
I'll leave it up to you.